plugins/carbon_black_defense/komand_carbon_black_defense/connection/connection.py

```diff
@@ -1,35 +1,83 @@
-import komand
+import insightconnect_plugin_runtime
+from insightconnect_plugin_runtime.exceptions import PluginException
+
 from .schema import ConnectionSchema, Input
-from komand.exceptions import ConnectionTestException
+from insightconnect_plugin_runtime.exceptions import ConnectionTestException
 
 # Custom imports below
 import requests
+from typing import Optional
+import json
 
 
-class Connection(komand.Connection):
+class Connection(insightconnect_plugin_runtime.Connection):
     def __init__(self):
         super(self.__class__, self).__init__(input=ConnectionSchema())
 
-    def connect(self, params):
+        self.job_id = None
+
+    def connect(self, params=None):
+        if params is None:
+            params = {}
         self.logger.info("Connect: Connecting..")
         self.host = params.get(Input.URL)
         self.token = params.get(Input.API_KEY).get("secretKey")
+        self.org_key = params.get(Input.ORG_KEY)
         self.connector = params.get(Input.CONNECTOR)
+        self.headers = {"X-Auth-Token": f"{self.token}/{self.connector}"}
+
+    def get_job_id_for_detail_search(self, event_id: str) -> Optional[str]:
+        response = self.call_api(
+            "POST",
+            f"{self.host}/api/investigate/v2/orgs/{self.org_key}/enriched_events/detail_jobs",
+            json_data={"event_ids": [event_id]},
+        ).get("job_id")
+        self.logger.info(f"The response is: {response}")
+        if response:
+            job_id = response
+            return job_id
+        return None
+
+    def check_status_of_detail_search(self, job_id: str = None):
+        response = self.call_api(
+            "GET",
+            f"{self.host}/api/investigate/v2/orgs/{self.org_key}/enriched_events/detail_jobs/" f"{job_id}",
+            json_data={"job_id": job_id},
+        )
+        self.logger.info(f"{response}")
+        contacted = response.get("contacted")
+        completed = response.get("completed")
+        if contacted and completed and (contacted == completed):
```

Comment:
Whats the purpose of the extra "==" check here?

Reply:
IIRC Mike mentioned it'd be good to check if contacted exists and if completed exists and if contacted and completed are equal

Reply:
> Ah- so contacted and completed aren't boolean values? Are they strings?

@pmara-r7 they are integers that come back, eg. `87`